packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -15,11 +15,17 @@ export function getTextMappings(
       try {
         mappings = JSON.parse(mappings);
       } catch (err) {
-        // fail gracefully
+        // fail gracefully - text mapping cannot be parsed
         console.error('Failure parsing text_mappings', err);
         mappings = null;
       }
+    } else {
+      // text mapping is not a string
+      mappings = null;
     }
+  } else {
+    // no text mappings are present
+    mappings = null;
```

Comment:
Instead of these 2 separate but identical else branches, I think it would be better to slightly refactor the code:

```
if (mappings && typeof mappings === 'string') {
    ...
} else {
    // text_mappings are not what we expect; reset to null
    mappings = null;
}
```